components/client/src/main/java/com/hotels/styx/client/netty/connectionpool/NettyToStyxResponsePropagator.java

```diff
@@ -95,9 +90,8 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
     private RuntimeException toStyxException(Throwable cause) {
         if (cause instanceof OutOfMemoryError) {
             return new StyxClientException("Styx Client out of memory. " + cause.getMessage(), cause);
-        } else {
-            return new BadHttpResponseException(origin, cause);
         }
+        return new BadHttpResponseException(origin, cause);
```

Comment:
I actually prefer the if {} else {} construct as it's a bit clearer - but you can argue it's more verbose for something that's obvious...